frontend-runitup/src/Components/Routes/ActiveRun.jsx

```diff
@@ -0,0 +1,223 @@
+          headers: getHeaders(),
+        }
+      );
+      if (!response.ok) {
+        throw new Error("Failed to complete run");
+      }
+      const data = await response.json();
+      message.success("Run completed successfully!");
+      // Redirect to the profile page
+      navigate("/profile");
+    } catch (error) {
+      message.error(`Error completing run: ${error.message}`);
+    }
+  };
+
+  if (!runData) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <div className="active-run-container">
+      <Title level={2} className="run-title">
+        Active Run
+      </Title>
+      <Row gutter={[16, 16]} className="stats-row">
+        <Col xs={24} sm={12} md={8}>
+          <Card className="stat-card">
+            <Statistic
+              title="Distance"
+              value={runData.distance}
+              precision={2}
+              suffix="miles"
+              prefix={<AimOutlined />}
+            />
+          </Card>
+        </Col>
+        <Col xs={24} sm={12} md={8}>
+          <Card className="stat-card">
+            <Statistic
+              title="Elapsed Time"
+              value={formatTime(elapsedTime)}
+              prefix={<ClockCircleOutlined />}
+            />
+          </Card>
+        </Col>
+        <Col xs={24} sm={12} md={8}>
+          <Card className="stat-card">
+            <Statistic
+              title="Target Pace"
+              value={formatPace(runData.averagePace)}
+              suffix="/mile"
+              prefix={<DashboardOutlined />}
+            />
+          </Card>
+        </Col>
+        <Col xs={24} sm={12} md={6}>
+          <Card className="stat-card">
+            <Statistic
+              title="Elevation Gain"
+              value={runData.elevationGain}
+              suffix="ft"
+              prefix={<RiseOutlined />}
+            />
+          </Card>
+        </Col>
+        <Col xs={24} sm={12} md={6}>
+          <Card className="stat-card">
+            <Statistic
+              title="Elevation Loss"
+              value={runData.elevationLoss}
+              suffix="ft"
+              prefix={<FallOutlined />}
+            />
+          </Card>
+        </Col>
+        <Col xs={24} sm={12} md={12}>
+          <Card className="stat-card">
+            <Statistic
+              title="Est. Calories Burned"
+              value={runData.estimatedCaloriesBurned}
+              suffix="cal"
+              prefix={<FireOutlined />}
+            />
+          </Card>
+        </Col>
+      </Row>
+      <Card className="map-card">
+        <div
+          ref={mapContainer}
+          style={{ width: "100%", height: "300px" }}
+        ></div>
+      </Card>
+      <Button
+        type="primary"
+        size="large"
+        onClick={handleCompleteRun}
+        className="complete-run-btn"
+      >
+        Complete Run
+      </Button>
+    </div>
+  );
+};
+
+const formatTime = (seconds) => {
+  const hours = Math.floor(seconds / 3600);
```

Comment:
I could be wrong but are there other places in the code where you are formatting time? I thought i remember seeing it. If so, can you make sure that there's one function defined for this and then you can reuse it throughout? Otherwise, feel free to disregard this comment

Reply:
You are correct. I will go back and do this in a bit!